app/controllers/admin/tutorials_controller.rb

```diff
@@ -3,7 +3,17 @@ def edit
     @tutorial = Tutorial.find(params[:id])
   end
 
-  def create; end
+  def create
+    videos = YouTube::Video.by_playlist(playlist_params[:playlist_id])
+    tutorial_info = YouTube::Playlist.info(playlist_params[:playlist_id])
```

Comment:
This looks like it breaks SRP but I am not sure. Could you explain the logic in this method's workflow & why this doesn't break SRP?